lib/hiera/backend/mysql_backend.rb

```diff
@@ -6,10 +6,22 @@ class Hiera
   module Backend
     class Mysql_backend
       def initialize
-        @use_jdbc = defined?(JRUBY_VERSION) ? true : false 
+        mysql_host=Config[:mysql][:host]
+        mysql_user=Config[:mysql][:user]
+        mysql_pass=Config[:mysql][:pass]
+        mysql_port=Config[:mysql][:port]
+        mysql_database=Config[:mysql][:database]
+
+        @use_jdbc = defined?(JRUBY_VERSION) ? true : false
         if @use_jdbc
           require 'jdbc/mysql'
           require 'java'
+          Jdbc::MySQL.load_driver
+          uri = "jdbc:mysql://#{mysql_host}:#{mysql_port}/#{mysql_database}"
+          props = java.util.Properties.new
+          props.set_property :user, mysql_user
+          props.set_property :password, mysql_pass
+          @conn = com.mysql.jdbc.Driver.new.connect(uri, props)
```

Comment:
@crayfishx I will test it later today. According to the driver it should happen automatically :).